docs/source/python/parquet.rst

```diff
@@ -210,6 +210,19 @@ Alternatively python ``with`` syntax can also be use:
 Data Type Handling
 ------------------
 
+Reading types as DictionaryArray
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``read_dictionary`` option in ``read_table`` and ``ParquetDataset`` will
+cause columns to be read as ``DictionaryArray``, which will become
+``pandas.Categorical`` when converted to pandas. This option is only valid for
```

Comment:
Is the limitation intended or simply because we only have it implemented for binary columns?

Reply:
It's only implemented for BYTE_ARRAY columns at the moment. We could expand that but there is little benefit from a performance/memory use point of view for the primitive types

Reply:
I've also used this (through pandas.Categorical) in the past on `date` and `float` types (e.g. in some datasets you can have 1000s of products that only have one of 5 prices). This often gave a 4-6x improvement in memory usage for these columns.

Reply:
(just dropping it here as FYI)

Reply:
I see. I'll open a JIRA as a follow up

Reply:
https://issues.apache.org/jira/browse/ARROW-6140